Documentation/LayoutCodeBehind.md

```diff
@@ -0,0 +1,67 @@
+---
+id: 11763499-79e9-4868-83e6-41f3061745d1
+title: "Layout CodeBehind"
+dateupdated: 2018-01-29
+---
+
+<a name="Overview" class="injected"></a>
+
+# Overview
+
+Xamarin.Android supports the auto generation of "Code Behind" classes. These classes
+can reduce the amount code a developer writes. You can end up replacing code like
+
+        var button = FindViewById<Button> (Resource.Id.myButton);
+        button.Click += delegate {
+        };
+
+with
+
+        myButton.Click += delegate {
+        };
+
+
+<a name="" class="injected"/></a>
+
+# Preparing to use Code Behind
+
+In order to make use of this new feature there are a few changes which are required. 
+An axml/xml file that you want to associate with an activity needs to be modified to 
+include a few extra xml attributes on the root layout element.
+
+    xmlns:tools="http://schemas.xamarin.com/android/tools"
+    tools:class="$(Namespace).$(ClassName)"
+
+The `class` attribute defines the Namespace and ClassName of the code which will be
+generated. For example if you have a layout for your `MainActivity` you would set
+the `tools:class` to `MyAppNamespace.MainActivity`. Note it should be the fully
+qualified name, not just the class name on its own.
+
+The next thing we need to do is to make the `MainActivity` a `partial` class. This
+allows the genereted code to extend the current class which you have written.
+So 
+    public class MainActivity : Activity {
+    }
+
+will become 
+
+    public partial class MainActivity : Activity {
+    }
+
+# How it works
+
+There are a couple of new MSBuild Tasks which generate the code behind.
+`CalculateLayoutCodeBehind` and `GenerateCodeBehindForLayout`. 
```

Comment:
If these are tasks, please use `<CalculateLayoutCodeBehind/>` and `<GenerateCodeBehindForLayout/>`. If these are targets, please use "the `CalculateLayoutCodeBehind` target", etc.